### PR TITLE
Versions edit button

### PR DESCRIPTION
**Pull request by vmcj**
image

I'm not really sure why the unit tests break (the do consistently so it has to be related) but do we want the button like this?

The unit test fails on `/jury/languages/c/edit` and when I try this locally it does work.
Make sure no contest is selected in the menu, visit the versions page, use the button is what the test should do and is what works for my side.

### CODE REVIEW

File: webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php [hunk 1]
```diff
@@ -224,7 +224,16 @@ public function visitWithNoContest(string $url, bool $dropdown): void
         $response = $this->client->getResponse();
         self::assertNotEquals(500, $response->getStatusCode(), sprintf('Failed at %s', $url));
         if ($dropdown && !str_contains($url, '/public')) {
-            self::assertSelectorExists('a#navbarDropdownContests:contains("no contest")');
+            try {
+                self::assertSelectorExists('a#navbarDropdownContests:contains("no contest")', "Failed at: " . $url);
+            } catch (\Exception $e) {
+                $gitlabArtifacts = getenv('GITLABARTIFACTS');
```
Comment (meisterT) [inline webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php:230]: Should we put this code into a test helper class?

Also, there is a bunch of phpcs errors here

Comment (vmcj) [reply webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php:230]: I think it's only relevant for the Traversal test so I'm not sure if we'll ever re use it.

File: webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php [hunk 2]
```diff
@@ -224,7 +224,16 @@ public function visitWithNoContest(string $url, bool $dropdown): void
...
+                if ($gitlabArtifacts != '') {
+                    $fileHandler = fopen(sprintf("%s/%s", $gitlabArtifacts, str_replace('/', '_s_', $url)) , 'w');
```
Comment (vmcj) [inline webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php:232]:
```suggestion
                    $fileHandler = fopen(sprintf("%s/%s", $gitlabArtifacts, str_replace('/', '_s_', $url)), 'w');
```

File: webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php [hunk 3]
```diff
@@ -224,7 +224,16 @@ public function visitWithNoContest(string $url, bool $dropdown): void
...
+                    fwrite($fileHandler, $response->getContent());
+		}
```
Comment (vmcj) [inline webapp/tests/E2E/Controller/ControllerRolesTraversalTest.php:234]:
```suggestion
                }
```

**Review by meisterT — APPROVED** [review]

**Pull request merged**